### ISSUE TITLE
Year drop down using <select>

### ISSUE THREAD

**Issue by iknowcss** [Hacker0x01/react-datepicker#462]
I'd really love to have an option to show the year drop down as a traditional `<select>` instead of the control that is there now. **I'm willing to implement and create a pull request**. I wanted to check first with the author and collaborators.
- Is this an acceptable feature?
- If so, what would you like the API to be?

The one situation that I don't know how to handle is the one where there is not a limit on both the upper and lower bounds of the date picker year. How would the group like to handle this?

Just taking a stab in the dark here at an api:

```
<DatePicker
    selected={this.state.startDate}
    onChange={this.handleChange} 
    showYearDropdown 
    yearDropdownType="select"
    dateFormatCalendar="MMMM" />
```

`yearDropdownType` options would be `select` and `scroll`, default to `scroll`.

**Comment by nvarchar** [comment]
This would be an excellent feature. Choosing historical dates, like birthdays, is currently pretty annoying.

**Comment by iknowcss** [comment]
@nvarchar I forked this repo, implemented the code with tests, and am using it on a project now. The only problem is that I didn't write generic CSS for it for this project. I will pull it up this week and see how hard it would be to merge upstream.

Would anyone else who wants this please +1 this so I can get a feel for interest?

**Comment by iknowcss** [comment]
(It's in a private repo)

**Comment by Faulik** [comment]
@iknowcss Hi, any news on this? It will be ok to even use your fork.

**Comment by iknowcss** [comment]
@Faulik @nvarchar sorry for not responding sooner. I have pushed the changes to a public fork here:
https://github.com/iknowcss/react-datepicker/tree/issue-462

This is branched off of the v0.25.0 release (tagged as such in the repo). I did a test rebase to the current Hacker0x01 master. There are a handful of conflicts in the actual source but it doesn't look too complicated to resolve. Unfortunately I don't have time to fix it myself right now.

The biggest block of work left to do is the generic styling. I didn't do it because the project I'm using it on had different styles anyway. I would have had to double-up.

Hope that helps!
